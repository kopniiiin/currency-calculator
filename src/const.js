export const Currency = {
  EUR: `EUR`,
  USD: `USD`,
  JPY: `JPY`
};

export const currenciesToTranslations = {
  [Currency.EUR]: `ÐµÐ²Ñ€Ð¾`,
  [Currency.USD]: `Ð´Ð¾Ð»Ð»Ð°Ñ€`,
  [Currency.JPY]: `Ð¸ÐµÐ½Ð°`
};

export const currenciesToEmoji = {
  [Currency.EUR]: `ðŸ’¶`,
  [Currency.USD]: `ðŸ’µ`,
  [Currency.JPY]: `ðŸ’´`
};

export const RATE_PRECISION = 2;
