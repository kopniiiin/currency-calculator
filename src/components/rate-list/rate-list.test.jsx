import React from "react";
import {shallow} from "enzyme";

import {Currency} from "../../const.js";

import RateList from "./rate-list.jsx";

describe(`snapshot test: RateList`, () => {
  it(`should render correctly`, () => expect(
    shallow(<RateList rates={[{currency: Currency.EUR}]}/>)
  ).toMatchSnapshot());

  it(`should render rate`, () => expect(
    shallow(<RateList rates={[{currency: Currency.EUR, rate: 4}]}/>)
  ).toMatchSnapshot());
});
