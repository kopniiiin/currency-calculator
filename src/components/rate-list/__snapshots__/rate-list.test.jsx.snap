// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: RateList should render correctly 1`] = `
<ul
  className=" rate-list"
>
  <li
    className="rate-list__item"
    key="EUR"
  >
    <span
      aria-label="ÐµÐ²Ñ€Ð¾"
      className="rate-list__currency"
    >
      ðŸ’¶
    </span>
     = 
    ?
    Â Ñ€ÑƒÐ±
  </li>
</ul>
`;

exports[`snapshot test: RateList should render rate 1`] = `
<ul
  className=" rate-list"
>
  <li
    className="rate-list__item"
    key="EUR"
  >
    <span
      aria-label="ÐµÐ²Ñ€Ð¾"
      className="rate-list__currency"
    >
      ðŸ’¶
    </span>
     = 
    4.00
    Â Ñ€ÑƒÐ±
  </li>
</ul>
`;
